tests/api.py

```
@@ -545,6 +549,229 @@ def test_regress(
             verbose=False,
         )
 
+    def test_fast_regress(
+        self,
+        nn_count,
+        batch_count,
+        variance_mode,
+        loss_method,
+        obj_method,
+        opt_method_and_kwargs,
+        nn_kwargs,
+        k_kwargs,
+    ):
+        target_mse, k_kwargs = k_kwargs
+        opt_method, opt_kwargs = opt_method_and_kwargs
+
+        if variance_mode is None:
+            sigma_method = None
+            apply_sigma_sq = False
+        else:
+            sigma_method = "analytic"
+            apply_sigma_sq = True
+
+        self._do_fast_regress_test_chassis(
+            train=self.train,
+            test=self.test,
+            target_mse=target_mse,
+            nn_count=nn_count,
+            batch_count=batch_count,
+            loss_method=loss_method,
+            obj_method=obj_method,
+            opt_method=opt_method,
+            sigma_method=sigma_method,
+            variance_mode=variance_mode,
+            nn_kwargs=nn_kwargs,
+            k_kwargs=k_kwargs,
+            opt_kwargs=opt_kwargs,
+            apply_sigma_sq=apply_sigma_sq,
+            verbose=False,
+        )
+
+
+class HeatonTest(FastRegressionAPITest):
+    @classmethod
+    def setUpClass(cls):
+        super(HeatonTest, cls).setUpClass()
+        with open(os.path.join(hardpath, heaton_file), "rb") as f:
+            cls.train, cls.test = pkl.load(f)
```

Did you actually run this file locally? This class name collides with the other `HeatonTest` and should throw an error. Please rename it or refactor `tests/api.py` into multiple files like `tests/regress_api.py`, `tests/classify_api.py`, and `tests/fast_regress_api.py`. If you do the latter, we will need to update the gitlab ci harness accordingly.